### DISCUSSION TITLE
Page Object Model with the "sb" fixture

### DISCUSSION THREAD

**Discussion by zh199609** [seleniumbase/SeleniumBase#2704] [General]
![image](https://github.com/seleniumbase/SeleniumBase/assets/32641509/998560b5-bf7c-4705-8483-7d0356076cd1)
The scope of SB fixture is function，Multiple testing methods will be repeated N times，
LoginPage(), I will also write N times，Is there a way to optimize this

Will it affect the operation of selenium base if I change the custom SB fixture range to session

**Comment by zh199609** [comment]
<img width="400" alt="image" src="https://github.com/seleniumbase/SeleniumBase/assets/32641509/b0ad6974-7838-488e-8a68-80c4ca8f2b1f">


I want to pass SB to the page layer in this way ,SB fixture has a small scope, code error
The failure came to an end

**Comment by mdmintz** [comment]
The 23 Syntax Formats are there to show you the allowable ways of structuring your tests. There's a lot of them to allow for flexibility depending on the user's needs, but the user needs to choose a specific format to work with. Note that `pytest` fixtures cannot be combined with the `BaseCase` formats, which means that if your `setup_page` method is in a class that inherits `BaseCase`, then it won't work because you're trying to add the `sb` `pytest` fixture.
